src/seedu/addressbook/AddressBook.java

```diff
@@ -383,12 +418,175 @@ private static String executeCommand(String userInputString) {
+	}
+
+    /**
+     * edits a persons 4 values, name, phone, email and school individually by taking input from a person 
+     *
+     * @param person - String[] representation of a person
+     * @return person - String[] representation of the new details of the person
+     */
+	private static String[] rebuildPerson(String[] person) {
+        showToUser(person);
+        final String[] personToAdd = new String[4];
+        personToAdd[0] = replaceUserDetails(person,"name",MESSAGE_TO_EDIT_PERSON_NAME,0);
+        personToAdd[1] = replaceUserDetails(person,"phone",MESSAGE_TO_EDIT_PERSON_PHONE,1);
+        personToAdd[2] = replaceUserDetails(person,"email",MESSAGE_TO_EDIT_PERSON_EMAIL,2);
+        personToAdd[3] = replaceUserDetails(person,"school",MESSAGE_TO_EDIT_PERSON_SCHOOL,3);
+        ALL_PERSONS.remove(person);     
+        addPersonToAddressBook(personToAdd);
+		return personToAdd;
+	}
+	
+    /**
+     * replaces a specific variable of a person, taking in user input
+     *
+     * @param person - String[] representation of a person, property - the specific name of the property to be changed
+     *                 help_message - the message to show the user when adding this variable, and finally an int type
+     *                 which is the type of variable it is - the type allows an automatic checking of the validity
+     *                 of the user input using the function  checkValidProperty(userInput,type)
+     *                 
+     * @return person - String[] representation of the new details of the person
+     */
+	private static String replaceUserDetails(String[] person, String property, String help_message,int type) {
+		final String tempProperty;
+		showToUser(help_message);
+        String userInput = getEditUserInput();
+        while (!checkValidProperty(userInput,type)&&!userInput.equals("")){
+        	showToUser("Please Enter Valid " + property);
+        	userInput = getEditUserInput();
+        }
+        if (userInput.equals("")){
+        	tempProperty = person[type];
+        } else {
+        	tempProperty = userInput;
+        }
+		return tempProperty;
+	}
+	
+    /**
+     * Checks if a user input is valid or not
+     *
+     * @param person - String userInput - what the user keyed in for the field, int - switch to the correct data validation check
+     *                 
+     * @return person - String[] representation of the new details of the person
+     */
+	private static boolean checkValidProperty(String property, int type) {
+		if (type == 0) {
+			return isPersonNameValid(property);
+		}
+		else if (type == 1) {
+			return isPersonPhoneValid(property);
+		}
+		else if (type == 2) {
+			return isPersonEmailValid(property);
+		}
+		else {
+			return isPersonSchoolValid(property);
+		}
+
+	}
+    /**
+     * looping function to find a unique person to edit, or else print an error message and take input again
```

Comment:
use active form of the verb in header comment.
this would rather be: `loops to find unique ...`
even better would be `Checks the validity of the input person ...` or something on those lines